include/host_object.h

```diff
@@ -0,0 +1,167 @@
+#pragma once
+
+#include "runtime.h"
```

Comment:
Is it our policy that all headers should explicitly include all their dependencies? I'm fine with that, but there's a lot of places where this is not enforced so far.

Reply:
If it's not then I would be in favor of making it so. At least that's what I've been trying to do. In my experience you otherwise often run into missing headers when compiling on a different platform (e.g. Windows, probably also libc++).

Reply:
I've addressed this for the new headers added in this PR,